### ISSUE TITLE
about pretraining dataset

### ISSUE THREAD

**Issue by Xixi7301210** [Abraham190137/TactileACT#4]
Hi, thank you for your work.
I am not sure about the size of your dataset for pertraining.  "To form the contrastive pairs for the pretraining, we sampled n (we used 7) observations randomly from a single demonstration". Does this mean that only 7 tactile images and 21 camera images were used in total？

**Comment by Abraham190137** [comment]
Sorry for the confusion. Here's the general pipeline: first, we collect demos, recording state information, actions (goal poses), and both the vision and tactile data (5 cameras + tactile data (244x244x3 "image" where the channels are strain_x, strain_y, strain_z (so float32)). It's a fairly standard IL training dataset; it just has tactile data too. 

Next, we do the pretraining. During this step, we use all of the training data. However, for the contrastive pretraining, we need to sample constrastive pairs. To generate each of these, we sample 7 timesteps from a single trajectory, and pull the images and tactile data for the timesteps. We then embed the images and tactile data, and train the network to align the embeddings of the tactile data to the embeddings of the images from the same timestep, and to make the tactile embeddings different from the image embeddings from different timesteps (ClIP loss). The data for these 7 timesteps is essential one "datapoint" for the contrastive loss - it's what the dataset's __getitem__ function returns. During training, we sample from the full dataset (all of the timesteps from all of the trajectories).
